components/brave_vpn/brave_vpn_service_desktop.cc

```diff
@@ -818,6 +877,10 @@ void BraveVpnServiceDesktop::SetPurchasedState(PurchasedState state) {
   ScheduleFetchRegionDataIfNeeded();
 }
 
+void BraveVpnServiceDesktop::OnSkusRemoteDisconnect() {
+  skus_service_.reset();
```

Comment:
you need to rebind after resetting if there is a connection error

Reply:
Started to look at how to do this in https://github.com/brave/brave-core/pull/10847/commits/bc08735f11dc33351fe3845cef9a29959b4e6eb4

Specifically see https://github.com/brave/brave-core/pull/10847/commits/bc08735f11dc33351fe3845cef9a29959b4e6eb4#diff-2170e2d6e88ab6e0202eac0280482f5a45f468d0fcc8d9d4d48fc358812b4a0cR882-R889

Reply:
also make sure you disconnect in the Shutdown method for KeyedService

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/3af3a81f930c0fcaaf61080214915786f3a070f5